File: packages/shared/src/project.ts
/**
 * Contratti condivisi per i progetti git.
 * Allineati con backend/src/app/models/project.py e schemas/project.py.
 */

export interface Project {
  id: number;
  name: string;
  repo_path: string;
  default_branch: string;
  test_command: string | null;
  active: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateProjectInput {
  name: string;
  repo_path: string;
  default_branch?: string;
  test_command?: string | null;
}

File: backend/src/app/models/project.py
"""Model ORM per i progetti git registrati.

Un Project rappresenta un repository locale su cui l'AI può lavorare i ticket di
codice (fix/feature): stacca un branch, applica le modifiche, esegue i test e — su
approvazione — committa.
"""

from datetime import datetime

from sqlalchemy import Boolean, DateTime, Integer, String, func
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class Project(Base):
    __tablename__ = "projects"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(120), nullable=False, unique=True)
    # Percorso assoluto del repository git locale.
    repo_path: Mapped[str] = mapped_column(String(1024), nullable=False)
    default_branch: Mapped[str] = mapped_column(String(120), nullable=False, default="main")
    # Comando di test eseguito dopo le modifiche (opzionale), es. "pytest -q".
    test_command: Mapped[str | None] = mapped_column(String(512), nullable=True)
    active: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)

    created_at: Mapped[datetime] = mapped_column(
        DateTime, server_default=func.now(), nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, server_default=func.now(), onupdate=func.now(), nullable=False
    )
